### PR TITLE
fix: Don't convert pydantic objects to TraceRecord at initial op call time.

### PR DESCRIPTION
**Pull request by shawnlewis**

### CODE REVIEW

File: weave/trace/vals.py [hunk 1]
```diff
@@ -122,21 +183,8 @@ def __getattribute__(self, __name: str) -> Any:
...
-        #    correct
-        # 2. The attribute is assigned as a property and is not bound at
-        #    assignment time. In this case, it is "unlikely" that the args
-        #    contain a `self` argument - which is why we apply this heuristic.
-        # 3. The attribute is assigned as a property and is bound to another
-        #    object at the time of assignment. In this case, the binding is
-        #    incorrect. However, in our evaluation use case we do not have this
-        #    case. We are accepting the incorrect assignment here for the sake
-        #    of expediency, but should be fixed.
         if isinstance(val_attr_val, Op) and inspect.signature(
```
Comment (tssweeney) [inline weave/trace/vals.py:188]: Dead code now?

**Review by tssweeney — APPROVED** [review]

File: weave/trace/vals.py (hunk 1 above)
Comment (shawnlewis) [reply weave/trace/vals.py]: Removed

**Review by tssweeney — CHANGES_REQUESTED** [review]
Hooman code is now failing on `evaluate.py`

```
Traceback (most recent call last):
  File "/Users/timothysweeney/Workspace/github/wandb/core/services/weave-python/weave-public/weave/flow/util.py", line 17, in process_item
    result = await func(item)
  File "/Users/timothysweeney/Workspace/github/wandb/core/services/weave-python/weave-public/weave/flow/eval.py", line 228, in eval_example
    raise e
  File "/Users/timothysweeney/Workspace/github/wandb/core/services/weave-python/weave-public/weave/flow/eval.py", line 226, in eval_example
    eval_row = await self.predict_and_score(example, model)
  File "/Users/timothysweeney/Workspace/github/wandb/core/services/weave-python/weave-public/weave/trace/op.py", line 74, in _run_async
    output = await awaited_res
  File "/Users/timothysweeney/Workspace/github/wandb/core/services/weave-python/weave-public/weave/flow/eval.py", line 132, in predict_and_score
    raise OpCallError(message)
weave.trace.errors.OpCallError: 
Call error: Error calling search: missing a required argument: 'k'

Options for resolving:
a. change QAModel.predict argument names to match a subset of dataset column names: channel, thread_id, message, ...
b. change dataset column names to match expected QAModel.predict argument names: ['message']
c. construct Evaluation with a preprocess_model_input function that accepts a dataset example and returns a dict with keys expected by QAModel.predict
```

**Review by tssweeney — APPROVED** [review]
LGTM! 🚢

**Pull request merged**